pkg/diagnostics/http_tracing.go

```diff
@@ -58,6 +58,12 @@ func HTTPTraceMiddleware(next fasthttp.RequestHandler, appID string, spec config
 			}
 		}
 
+		// Check if response has traceparent header and add if absent
```

Comment:
you can take this to helper method UpdateSpanContextInResponse..and in that method, you check for header and update the header..

Reply:
We do not use that logic anywhere again so is it worth moving it to helper method? I am thinking to remove the checkResponseHeader helper as well as we are not reusing it anywhere.